src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -100,6 +103,120 @@ public bool HideAutocompleteMethodOptions
             }
         }
 
+        private IEnumerable<ClusterAutocompleteResult> clusterResults;
+        /// <summary>
+        /// Cluster autocomplete search results.
+        /// </summary>
+        public IEnumerable<ClusterAutocompleteResult> ClusterResults
+        {
+            get
+            {
+                return clusterResults;
+            }
+            set
+            {
+                clusterResults = value;
+                RaisePropertyChanged(nameof(ClusterResults));
+                RaisePropertyChanged(nameof(NthofTotal));
+                RaisePropertyChanged(nameof(ResultsLoaded));
+                RaisePropertyChanged(nameof(ConfirmSource));
+                RaisePropertyChanged(nameof(PreviousSource));
+                RaisePropertyChanged(nameof(NextSource));
+            }
+        }
+
+        public bool ResultsLoaded => ClusterResults != null;
+        public bool IsOpen { get; set; }
+
+        private int ClusterResultsCount => ClusterResults == null ? 0 : ClusterResults.Count();
+
+        private int selectedIndex = 0;
+        /// <summary>
+        /// Cluster autocomplete search results.
+        /// </summary>
+        public int SelectedIndex
+        {
+            get
+            {
+                return selectedIndex;
+            }
+            set
+            {
+                if(selectedIndex != value && value >= 0)
+                {
+                    ReAddNode(value);
+                }
+                selectedIndex = value;
+                RaisePropertyChanged(nameof(SelectedIndex));
+                RaisePropertyChanged(nameof(NthofTotal));
+                RaisePropertyChanged(nameof(PreviousSource));
+                RaisePropertyChanged(nameof(NextSource));
+            }
+        }
+
+        private void ReAddNode(int index)
+        {
+            if(fullResults == null)
+            {
+                return;
+            }
+            var results = fullResults.Results.ToList();
+            if(index >=  0 && index  < results.Count)
+            {
+                AddCluster(results[index]);
+            }
+        }
+
+        public void ConsolidateTransientNodes()
```

Comment:
Would you add some comments for the public functions?

Reply:
Converted to internal